app/controllers/api/v1/interactive_pages_controller.rb

```diff
@@ -220,13 +220,35 @@ def create_page_item
     when /LibraryInteractive/
       library_interactive = LibraryInteractive.find_by_serializeable_id(embeddable_type)
       return error("Invalid page_item[embeddable] parameter") if library_interactive.nil?
+
       embeddable = ManagedInteractive.create!(library_interactive_id: library_interactive.id)
     when /MwInteractive/
       embeddable = MwInteractive.create!
     when /Embeddable::Xhtml/
       embeddable = Embeddable::Xhtml.create!
+    when /Plugin_(\d+)::windowShade/
+      # Parse the embeddable_type string to get the approved_script id
+      # as well as the plugin_type
+      # For windowshade items we need to create a new embeddable of type
+      # Embeddable::Plugin and then associate it with the plugin.
+      # 1. Create a plugin with the instance of the approved_script
+      # 2. Create a Embeddable::EmbeddablePlugin
+      # Example: "Plugin_10::windowShade"
+      # IMPORTANT: 'windowShade' as the component_label is critical.
+      tip_type = 'windowShade'
+      regex = /Plugin_(\d+)::windowShade/
+      script_id = embeddable_type.match(regex)[1]
+      author_data = { tipType: tip_type }.to_json
+      # I am following the convention I saw in interactive_pages_controller.rb
+      embeddable = Embeddable::EmbeddablePlugin.create!
+      embeddable.approved_script_id = script_id
+      embeddable.author_data = author_data
+      embeddable.component_label = tip_type
+      embeddable.is_half_width = false
+      embeddable.save!
+
     else
-      return error("Only library interactive embeddables, iFrame interactives, and text blocks are currently supported")
+      return error("Unknown embbeddable_type: #{embeddable_type}\nOnly library interactive embeddables, iFrame interactives, and text blocks are currently supported")
```

Comment:
Not of critical importance, but should teacher edition windowshades be added to the list of supported embeddables? Or maybe at this point, we don't need to list what's supported since everything we plan on supporting is now supported?